packages/flutter_tools/lib/src/debug_adapters/flutter_test_adapter.dart

```diff
@@ -19,7 +19,7 @@ import 'mixins.dart';
 
 /// A DAP Debug Adapter for running and debugging Flutter tests.
 class FlutterTestDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments, FlutterAttachRequestArguments>
```

Comment:
Out of curiosity, any reason this doesn't implement `FlutterDebugAdapter`?

Reply:
It probably could, but it would inherit some stuff that doesn't make sense for tests (eg. code that spawns `flutter run` processes).

Perhaps we should have an additional `abstract` base class for both Flutter debug adapters that has the shared stuff (but none of the specific stuff). I'll make a note to have a look at this and see if it can be cleaned up a little (in a separate PR).

Reply:
Ahh, true, if you just implemented it, you might have to reproduce some functionality. And then extending you might over-inherit.